contracts/rewards/StableUsnMaker.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+// P1 - P3: OK
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/ISwap.sol";
+
+contract StableUsnMaker is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    ISwap public stableSwap;
+    address public lpMaker;
+    address private immutable usn;
+    address private immutable usdc;
+    address private immutable usdt;
+
+    event LogSetLPMaker(address oldLPMaker, address newLPMaker);
+    event LogSetStableSwap(address oldStableSwap, address newStableSwap);
+
+    event LogWithdrawFees();
+
+    event LogSwapStableTokenToUsn(address stableTokenToConvertToUsn, uint256 stableTokenAmount);
+
+    event LogUsnSentToLPMaker(uint256 usnAmount);
+
+    constructor(
+        address _stableSwap,
+        address _lpMaker,
```

Comment:
if we swap stable fees USDC/USDT into USN and then create 3Pool LPs using only USN as singleSideLiquidity, it will be very lossy
more efficient way is to directly supply USDC,USDT & USN earned via fees directly into 3pool as liquidity imbalance.